Bottom: b10774af614d756988a15191dee45e98d24552c1
Top:    4f32d9c6c8c00aeeec0970657e0cc03993708c11
Author: Ian Kent <raven@themaw.net>
Date:   2013-07-31 14:54:39 +0800

autofs-5.0.7 - teach dumpmaps to output simpe key value pairs


---

diff --git a/daemon/automount.c b/daemon/automount.c
index 6849272..c52eedc 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -2125,21 +2125,23 @@ int main(int argc, char *argv[])
 			program);
 #endif
 
-	if (argc == 0)
-		master_list = master_new(NULL, timeout, ghost);
-	else
-		master_list = master_new(argv[0], timeout, ghost);
-
-	if (!master_list) {
-		printf("%s: can't create master map %s", program, argv[0]);
-		exit(1);
-	}
-
 	if (dumpmaps) {
 		struct master_mapent *entry;
 		struct list_head *head, *p;
 		struct mapent_cache *nc;
 
+		/*
+		 * We can't specify a master map on the command line
+		 * when listing one or more maps for display so the
+		 * only way to alter it from the ddefault name is to
+		 * set ti in the configuration.
+		 */
+		master_list = master_new(NULL, timeout, ghost);
+		if (!master_list) {
+			printf("%s: can't create master map", program);
+			exit(1);
+		}
+
 		open_log();
 
 		master_init_scan();
@@ -2168,6 +2170,16 @@ int main(int argc, char *argv[])
 		exit(0);
 	}
 
+	if (argc == 0)
+		master_list = master_new(NULL, timeout, ghost);
+	else
+		master_list = master_new(argv[0], timeout, ghost);
+
+	if (!master_list) {
+		printf("%s: can't create master map %s", program, argv[0]);
+		exit(1);
+	}
+
 	become_daemon(foreground, daemon_check);
 
 	if (pthread_attr_init(&th_attr)) {
